app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -114,11 +114,24 @@ def verify_credentials_for_rhevm(options = {})
     raise e
   rescue RestClient::Unauthorized
     raise MiqException::MiqInvalidCredentialsError, "Incorrect user name or password."
+  rescue OvirtSDK4::Error => e
+    rethrow_as_a_miq_error(e)
   rescue
     _log.error("Error while verifying credentials #{$ERROR_INFO}")
     raise MiqException::MiqEVMLoginError, $ERROR_INFO
   end
 
+  def rethrow_as_a_miq_error(e)
+    if e.message.include?("The username or password is incorrect")
```

Comment:
I'd suggest to write here a comment (maybe a TODO) explaining that in version 4.2 of the SDK there will be specific exceptions for these cases. Once we update to 4.2 this should be modified to use those exceptions instead of this fragile string checking.

Reply:
I will create a git issue once this is merged, I think its easier to track than a TODO in the code.
Also added a trello card in backlog.